packages/gestalt/src/contexts/Theme.js

```diff
@@ -0,0 +1,132 @@
+// @flow strict
+import * as React from 'react';
+import PropTypes from 'prop-types';
+
+export type ColorScheme = 'light' | 'dark' | 'userPref';
+
+export const ColorSchemePropType: React$PropType$Primitive<ColorScheme> = PropTypes.oneOf(
+  ['light', 'dark', 'userPref']
+);
+
+type Theme = {|
+  name: string,
+  colorRed0: string,
+  colorRed100: string,
+  colorGray0: string,
+  colorGray50: string,
+  colorGray100: string,
+  colorGray200: string,
+  colorGray300: string,
+  colorGray400: string,
+|};
+
+type Props = {|
+  children: React.Node,
+  colorScheme?: ColorScheme,
+|};
+
+const lightModeTheme = {
+  name: 'lightMode',
+  colorRed0: '#ff5247',
+  colorRed100: '#e60023',
+  colorGray0: '#fff',
+  colorGray50: '#fff',
+  colorGray100: '#efefef',
+  colorGray200: '#767676',
+  colorGray300: '#111',
+  colorGray400: '#000',
+};
+
+const darkModeTheme = {
+  name: 'darkMode',
+  colorRed0: '#e60023',
+  colorRed100: '#ff5247',
+  colorGray0: '#050505',
+  colorGray50: '#272727',
+  colorGray100: '#494949',
+  colorGray200: '#b8b8b8',
+  colorGray300: '#efefef',
+  colorGray400: '#fff',
+};
+
+const ThemeContext: React.Context<Theme> = React.createContext<Theme>(
+  lightModeTheme
+);
+
+const themeToStyles = theme => {
+  let styles = '';
+  Object.keys(theme).forEach(key => {
+    if (key.startsWith('color')) {
+      styles += `  --${key}: ${theme[key]};\n`;
+    }
+  });
+  return styles;
+};
+
+let themeId = 0;
+const getThemeId = () => {
+  themeId += 1;
```

Comment:
@jennyscript could you explain why we need a global `themId` which increments each time we call `getThemeId`? Could we think of an alternative?

Reply:
I added this to make sure each ThemeProvider has a unique className. This is needed so you can nest themes or use different themes in parallel on the same page. It's kind of an edge case, but seemed like a reasonable thing to support. I could also try setting it up with a timestamp or other way to get uniqueness if this approach isn't ok

Reply:
Perhaps this namespace could be passed  in for the moment? Thinking more about it, we might have issues with SSR not hydrating correctly + it might be difficult to apply to the `<html>` element. `<ThemeProvider namespace=":root">` and make that the default in the `GestaltProvider`.

Reply:
I like that idea. Trying it out...

Reply:
This made things much more clear, thanks. I called it `id` to be a little more standard.